fabric-object-builders/src/main/java/net/fabricmc/fabric/api/block/FabricBlockSettings.java

```diff
@@ -43,56 +39,70 @@
 	static {
 		BlockConstructedCallback.EVENT.register(FabricBlockSettings::onBuild);
 	}
 
-	protected final Block.Settings delegate;
-	private final FabricBlockSettingsDelegate castDelegate;
+    private static final Map<Block.Settings, ExtraData> EXTRA_DATA = new HashMap<>();
 
-	private static class MiningInformation {
-		private final Tag<Item> tag;
-		private final int miningLevel;
+    private final Block.Settings delegate;
+	private final BlockSettingsHooks hooks;
 
-		MiningInformation(Tag<Item> tag, int miningLevel) {
-			this.tag = tag;
-			this.miningLevel = miningLevel;
-		}
-	}
+	static final class ExtraData {
+        private final List<MiningLevel> miningLevels = new ArrayList<>();
+        /* @Nullable */ private Boolean breakByHand;
+
+		private ExtraData(Block.Settings settings) {
+        }
+
+		void breakByHand(boolean breakByHand) {
+		    this.breakByHand = breakByHand;
+        }
 
-	private static class Information {
-		private Boolean breakByHand;
-		private List<MiningInformation> miningInformation = new ArrayList<>();
+		void addMiningLevel(Tag<Item> tag, int level) {
+		    miningLevels.add(new MiningLevel(tag, level));
+        }
 	}
 
-	private static Map<Block.Settings, Information> info = new HashMap<>();
+    private static final class MiningLevel {
+        private final Tag<Item> tag;
+        private final int level;
+
+        MiningLevel(Tag<Item> tag, int level) {
+            this.tag = tag;
+            this.level = level;
+        }
+    }
+
+	static ExtraData computeExtraData(Block.Settings settings) {
+	    return EXTRA_DATA.computeIfAbsent(settings, ExtraData::new);
+    }
 
 	private static void onBuild(Block.Settings settings, Block block) {
 		// TODO: Load only if fabric-mining-levels present
-		Information i = info.get(settings);
-		if (i != null) {
-			if (i.breakByHand != null) {
-				ToolManager.entry(block).setBreakByHand(i.breakByHand);
+		ExtraData data = EXTRA_DATA.get(settings);
+		if (data != null) {
+			if (data.breakByHand != null) {
+				ToolManager.entry(block).setBreakByHand(data.breakByHand);
 			}
-
-			for (MiningInformation mi : i.miningInformation) {
-				ToolManager.entry(block).putBreakByTool(mi.tag, mi.miningLevel);
+			for (MiningLevel tml : data.miningLevels) {
+				ToolManager.entry(block).putBreakByTool(tml.tag, tml.level);
 			}
 		}
 	}
 
-	protected FabricBlockSettings(Material material, MaterialColor color) {
+	private FabricBlockSettings(Material material, MaterialColor color) {
 		this(Block.Settings.of(material, color));
 	}
 
-	protected FabricBlockSettings(Block base) {
+	private FabricBlockSettings(Block base) {
 		this(Block.Settings.copy(base));
 	}
 
-	protected FabricBlockSettings(final Block.Settings delegate) {
+	private FabricBlockSettings(final Block.Settings delegate) {
 		this.delegate = delegate;
-		castDelegate = (FabricBlockSettingsDelegate) delegate;
+		hooks = (BlockSettingsHooks) delegate; // TODO: Redundant optimization?
```

Comment:
It's more about making later code cleaner. The object is meant to be short-lived anyway.

Reply:
Fair, I can remove the comment if the cached reference is to be retained